benchmark/sparse_blas/sparse_blas.cpp

```diff
@@ -0,0 +1,498 @@
+            // randomly permute column indices
+            std::vector<itype> cols(size2[1]);
+            std::iota(cols.begin(), cols.end(), 0);
+            for (gko::size_type row = 0; row < size2[0]; ++row) {
+                std::shuffle(cols.begin(), cols.end(), rng);
+                for (int i = 0; i < local_rowlength; ++i) {
+                    data.nonzeros.emplace_back(
+                        row, cols[i],
+                        gko::detail::get_rand_value<etype>(val_dist, rng));
+                }
+            }
+            data.ensure_row_major_order();
+            mtx2_ = Mtx::create(exec, size2);
+            mtx2_->read(data);
+        } else if (mode_str == "dense") {
+            const auto size2 = gko::dim<2>(size[1], FLAGS_spgemm_rowlength);
+            // don't expect too much quality from this seed =)
+            std::default_random_engine rng(FLAGS_seed);
+            std::uniform_real_distribution<gko::remove_complex<etype>> dist(
+                -1.0, 1.0);
+            gko::matrix_data<etype, itype> data{size2, dist, rng};
+            data.ensure_row_major_order();
+            mtx2_ = Mtx::create(exec, size2);
+            mtx2_->read(data);
+        } else {
+            throw gko::Error{__FILE__, __LINE__,
+                             "Unsupported SpGEMM mode " + mode_str};
+        }
+    }
+
+    std::pair<bool, double> validate() const override
+    {
+        auto ref = gko::ReferenceExecutor::create();
+        auto correct = Mtx::create(ref, mtx_out_->get_size());
+        gko::clone(ref, mtx_)->apply(mtx2_.get(), correct.get());
+        return validate_result(correct.get(), gko::clone(ref, mtx_out_).get());
+    }
+
+    gko::size_type get_flops() const override
+    {
+        auto host_mtx = Mtx::create(mtx_->get_executor());
+        auto host_mtx2 = Mtx::create(mtx_->get_executor());
+        host_mtx->copy_from(mtx_);
+        host_mtx2->copy_from(mtx2_.get());
+        // count the individual products a_ik * b_kj
+        gko::size_type work{};
+        for (gko::size_type row = 0; row < host_mtx->get_size()[0]; row++) {
+            auto begin = host_mtx->get_const_row_ptrs()[row];
+            auto end = host_mtx->get_const_row_ptrs()[row + 1];
+            for (auto nz = begin; nz < end; nz++) {
+                auto col = host_mtx->get_const_col_idxs()[nz];
+                auto local_work = host_mtx2->get_const_row_ptrs()[col + 1] -
+                                  host_mtx2->get_const_row_ptrs()[col];
+                work += local_work;
+            }
+        }
+        return 2 * work;
+    }
+
+    gko::size_type get_memory() const override
+    {
+        // read and write everything only once, ignore row pointers
+        return (mtx_->get_num_stored_elements() +
+                mtx2_->get_num_stored_elements() +
+                mtx_out_->get_num_stored_elements()) *
+               (sizeof(etype) + sizeof(itype));
+    }
```

Comment:
Nice idea, we should maybe think of something like this for other benchmarks as well.

Reply:
#802 *wink wink*